### PR TITLE
Run ES self provisioned e2e test on travis using OCP 3.9

### LINKED ISSUE #484: Run self es-self provisioned test on travis
Run `make e2e-tests-self-provisioned-es` on travis. The motivation is to catch issues early.

For example https://github.com/jaegertracing/jaeger-operator/issues/481 is not related to OCP version but to change in ES operator.

@kevinearls would you be interested in this issue?

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #484

### CODE REVIEW

File: deploy/role.yaml [hunk 1]
```diff
@@ -43,6 +43,10 @@ rules:
 - apiGroups:
   - extensions
   resources:
+  - replicasets
```
Comment (pavolloffay) [inline deploy/role.yaml]: These are required to run on OCP 3.9

File: deploy/role.yaml [hunk 2]
```diff
@@ -79,6 +83,7 @@ rules:
   - '*'
 - apiGroups:
   - apps
+  - extensions
```
Comment (pavolloffay) [inline deploy/role.yaml]: Required to run on OCP 3.9

**Comment by pavolloffay** [comment]
travis is yellow, but when you click on it it is all green. Maybe travis reporting bug.

File: .travis/setupOpenShift.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+set -eoux pipefail
+
+export OPENSHIFT_VERSION=v3.9
```
Comment (jpkrohling) [inline .travis/setupOpenShift.sh]: We should start with 4.1, and I think there's no `oc cluster up` for that.

Comment (pavolloffay) [reply .travis/setupOpenShift.sh]: I don't think 4.1 will run on travis, it requires full virtualization and way more resources. This is the only way we can get it working here.

Other option is 3.11, but I prefer 3.9 due to a quick start and low resources utilization.

**Comment by pavolloffay** [comment]
I have not seen issues with the streaming anymore, although it should not be related to this PR.

**Pull request merged**